Sync Update

flossypurse opened the pull request:
WIP

mmcshane commented on sync_update/your_updatable_workflow_dacx.go:
```diff
@@ -0,0 +1,157 @@
+// Updates can be sent to the Workflow during this time.
+func YourUpdatableWorkflow(ctx workflow.Context, param WFParam) (WFResult, error) {
+	counter := param.StartCount
+	workflow.SetUpdateHandler(ctx, YourUpdateName, func(arg YourUpdateArg) (YourUpdateResult, error) {
```
Should check the returned error here.

mmcshane commented on sync_update/your_updatable_workflow_dacx.go:
```diff
@@ -0,0 +1,157 @@
+
+// isPositive is a validator function.
+// It returns an error if the int value is below 1.
+func isPositive(ctx workflow.Context, u YourUpdateArg) error {
```
Strictly speaking the workflow.Context parameter is optional for both the validation function and the execution function. And if it's present in one but not the other, that's fine. But from an example standpoint it's probably more clear if it's either in both or neither.

mmcshane reviewed (changes requested):
Couple of minor things but otherwise this looks good.

flossypurse commented:
@mmcshane addressed all of your comments - but I changed to a new branch -
Pushed it all to this one: https://github.com/temporalio/documentation-samples-go/pull/17
Closing this in favor of the other one.
